solidity/test/StandardPoolConverter.js

```diff
@@ -1210,4 +1148,394 @@ describe('StandardPoolConverter', () => {
+                                await addLiquidity(reserveToken1, reserveToken2, converter, [initialBalance1, initialBalance2].map(n => ONE_TOKEN.muln(n)));
+
+                                const conversionAmount = ONE_TOKEN.muln(Math.max(initialBalance1, initialBalance2));
+                                const conversion = await convert(reserveToken1, poolToken, reserveToken2, bancorNetwork, converter, conversionAmount);
+                                const expectedFeeBase = conversion.fee.muln(networkFeePercent).divn(200);
+                                const reserveBalance1 = ONE_TOKEN.muln(initialBalance1).add(conversionAmount);
+                                const reserveBalance2 = ONE_TOKEN.muln(initialBalance2).sub(conversion.amount);
+
+                                const supplyAmount = await poolToken.balanceOf(defaultSender);
+                                await removeLiquidity(reserveToken1, reserveToken2, converter, supplyAmount, true);
+
+                                const expectedFee1 = expectedFeeBase.mul(reserveBalance1).div(reserveBalance2);
+                                const expectedFee2 = expectedFeeBase;
+
+                                const actualFee1 = await reserveToken1.balanceOf(networkFeeWallet);
+                                const actualFee2 = await reserveToken2.balanceOf(networkFeeWallet);
+
+                                expectAlmostEqual(actualFee1, expectedFee1, '0', '0.02383');
+                                expectAlmostEqual(actualFee2, expectedFee2, '0', '0.02383');
+                            });
+                        }
+                    }
+                }
+            }
+
+            function description(prefix, initialBalance1, initialBalance2, conversionFeePercent, networkFeePercent) {
+                return prefix
+                    + ` initial balances = [${initialBalance1}, ${initialBalance2}],`
+                    + ` conversion fee = ${conversionFeePercent}%`
+                    + ` and network fee = ${networkFeePercent}%`
+                ;
+            }
+
+            async function createPool(networkFeePercent, conversionFeePercent, ethIndex) {
+                const poolToken = await DSToken.new('poolToken', 'poolToken', 18);
+                const reserveToken1 = await TestStandardToken.new('reserveToken1', 'reserveToken1', 18, TOTAL_SUPPLY);
+                const reserveToken2 = await TestStandardToken.new('reserveToken2', 'reserveToken2', 18, TOTAL_SUPPLY);
+                const converter = await StandardPoolConverter.new(poolToken.address, contractRegistry.address, 1000000);
+
+                const reserveToken = [{}, reserveToken1, reserveToken2][ethIndex];
+                reserveToken.address = NATIVE_TOKEN_ADDRESS;
+                reserveToken.approve = async (spender, value) => {};
+                reserveToken.balanceOf = async (account) => new BN(await web3.eth.getBalance(account));
```

Comment:
I don't understand the purpose of `reserveToken` here. It is never used.

Reply:
@sshmaxime please ignore this. I've already rewritten lots of this code in an upcoming PR.

Reply:
It points to `reserveToken1` when ethIndex == 1`, to `reserveToken2` when `ethIndex == 2`, and to an unused object when `ethToken == 0`.